### DISCUSSION TITLE
Beginner help to compiling open source software<title>

### DISCUSSION THREAD

**Discussion by psyfoused** [community/community#130547] [Programming Help]
### Select Topic Area

Question

### Body

I apologize in advance, but I have been tinkering for sometime with open source software and unfortunately have had zero success compiling these repositories into working software. again, my apologies if this has been addressed in the forum before. I've read through numerous tutorials and pages on this topic, but I am not a programmer. I do want to take this on as a hobby and have collected dozens of projects to build great programs, but still never managed to git(pun) any of them going. Is there a step by step process that is understandable for beginners that someone could please link me too? a lot of this stuff reads like Charlie Brown to me but I am confident with the right guidance, I can get this going. Any direction and help would be so amazing as I could only imagine how busy a lot of you must be. And, on a more specific note: for a few months there I was using a Collab Notebook project that involved Text to video generative AI.
The program worked great for a while until I received a "Pytorch missing-error". of course there was some vague info on how to fix or edit the problem, but to no avail. 
Note: I have gone through the process of using my Terminal to add commands for the necessary Github software and installing pre-reqs, but nothing seems to  have helped. I am familiar with the Terminal and have used it in the past. It seems what i am looking for is an all in one, "drag and drop" compiler that produces a ".dmg"  file so i can easily install these softwares i am so eager to use. My apologies to anyone and everyone for this rudimentary information inquery. I do appreciate the time and thank you for your patience. I hope all are well and hope everyone peace and prosperity. Much Love.

**Comment by asdacosta** [comment]
This guide to compiling open-source software: https://opensource.com/article/21/11/compiling-code


Also, regarding your PyTorch Error, you could had spent more time searching for the solution instead of giving up. Know that you’re not the first person encountering such error, so there is definitely the solution online and you’d have to alter your searching methods before you’ll find it
Giving up on errors is detrimental, as you can do so in subsequent projects.

**Comment by psyfoused** [comment]
hey Abraham, i sincerely appreciate the link you sent me. to my defense, i
didnt exactly give up. the error had a direct link to fix the problem.
however after trying for several attempts, the edit was not going through.
whatever the case, something i was doing was wrong. programmers may have a
harder time understanding these obviously easy answers, but lets say you
are a cave man in ancient times and a time traveler comes from the future
and gives you a motorcycle with a full tank of gas. now, as a caveman, i
can get that thing to turn on and i can ride it. but it will take about
10,000 years before anyone will be able to put gas back in it. thanks for
your time and advice.

**Comment by asdacosta** [comment]
You’re welcome. Kindly mark my comment as the answer if it was helpful.

**Comment by psyfoused** [comment]
Good morning, just to follow up, I used the link and instructions it
provided and from the first command i typed in i received  a "command not
found" message. Compiling open source software is extremely difficult and I
am clearly doing something wrong. I appreciate the help but unfortunately
your answer has not solved the problem.

**Comment by psyfoused** [comment]
Hey good morning, so this is where Im at: Ive been trying to install
Homebrew. Ive gotten as far as the install but i get held up here:
at this "next two steps" part. it says :

*==> Next steps:*

*- Run these two commands in your terminal to add Homebrew to your `$PATH`:
    (echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)"') >> ~/.zprofile
    eval "$(/opt/homebrew/bin/brew shellenv)"*

now this where I always get hung up. Idk if im not copying and pasting
correctly or what. In fact, I dont even know where the two commands begin,
separate , or end. Meaning; where does the command start ? am i supposed to
copy the '$PATH`:  word too? where do these two commands split? run them
together? this is the stuff that is so vague and Im just trying to do the
simplest thing and install Homebrew. I'd love to give you the credit for
answering for me. At this point im super committed to figuring this out. I
have done the follow ups of uninstalling Homebrtew, trying to address the
error messages and what not. I keep doing all the things and it's stil
saying:
 "zsh: command not found"
meaning Homebrew is not installed. And to be clear my dude, I am not a
quitter. Quite the opposite. I would surely appreciate some help or
guidance. Thank you . I hope you are not to busy or I am interfering with
your private life. Thanks again either way.

Sean Slater

**Reply by asdacosta** [reply]
The message is showing you how to add Homebrew to your system's PATH such that you can run the Homebrew commands from any directory. 
This is what you’re supposed to paste and it actually contains two commands:

`echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)"' >> ~/.zprofile; eval "$(/opt/homebrew/bin/brew shellenv)" `

**Comment by psyfoused** [comment]
thank you for getting back. the commands produced these results:

zsh: parse error near `\n'

***@***.*** ~ % ~/.zprofile; eval
"$(/opt/homebrew/bin/brew shellenv)"



zsh: permission denied: /Users/universalconcepts/.zprofile

***@***.*** ~ %


ive even been asking chat gpt.


these are the files i want to instal : "__init__.py, attend.py,
make_a_video.py, and setup.py "

how do i just compile these files into the working software it claims to be?

**Reply by asdacosta** [reply]
The semicolon separating the two commands might be causing the error near ‘\n’ so you’ll need to paste them separately. Also, prepend `sudo` to overcome the permission error. I have done each for you so paste them separately:

`sudo echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)"' >> ~/.zprofile`

`eval "$(/opt/homebrew/bin/brew shellenv)"`

**Reply by asdacosta** [reply]
If you also want install Python packages use:
`pip install the_package_name`

**Reply by ebndev** [reply]
@asdacosta thanks for taking the time to answer and being a part of the GitHub Community! 😄
